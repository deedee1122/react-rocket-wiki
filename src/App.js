import React, { useState, useEffect } from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import "./App.css";
import Navbar from "./components/Navbar";
import Home from "./pages/Home";
import LaunchDetails from "./pages/LaunchDetails";
import RocketDetails from "./pages/RocketDetails";
import Rockets from "./pages/Rockets";
import axios from "axios";

function App() {
  let base = "https://api.spacexdata.com/v3/";
  let [result, setResult] = useState([]);
  let [limit, setLimit] = useState(10);
  let [offset, setOffset] = useState(0);
  // console.log(result);
  let [search, setSearch] = useState("");
  let api = `${base}launches?limit=${limit}&offset=${offset}`;

  let [rocketResult, setRocketResult] = useState([]);
  let rocketAPI = `${base}rockets?limit=${limit}&offset=${offset}`;

  useEffect(() => {
    let fetchData = axios
      .get(api)
      .then((res) => setResult(res?.data))
      .catch((err) => console.log(err));

    let fetchRocketData = axios
      .get(rocketAPI)
      .then((res) => setRocketResult(res?.data))
      .catch((err) => console.log(err));
  }, [api, rocketAPI]);

  return (
    <Router>
      <Navbar />
      <Routes>
        <Route
          path="/"
          element={<Home setOffset={setOffset} result={result} />}
        />
        <Route
          path="/launches/:id"
          element={<LaunchDetails result={result} />}
        />
        <Route
          path="/rockets"
          element={<Rockets rocketResult={rocketResult} />}
        />
        <Route
          path="/rockets/:id"
          element={<RocketDetails rocketResult={rocketResult} />}
        />
      </Routes>
    </Router>
  );
}

export default App;
