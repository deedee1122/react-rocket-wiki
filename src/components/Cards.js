import React from "react";
import { Link } from "react-router-dom";

const Cards = ({
  links,
  mission_name,
  rocket,
  launch_success,
  flight_number,
}) => {
  return (
    <Link
      to={`/launches/${flight_number}`}
      data-testid="launch-cards"
      className="card col-lg-2 col-md-3 col-11"
    >
      <style jsx="true">
        {`
          .card {
            padding: 0;
            position: relative;
          }
          a {
            text-decoration: none !important;
            color: #212529 !important;
          }
          .badge {
            position: absolute;
            top: 10px;
            right: 10px;
          }
        `}
      </style>
      <img
        src={links?.mission_patch_small}
        className="card-img-top"
        alt="..."
      />

      <div className="card-body">
        <h5 className="card-title">{mission_name}</h5>
        <p className="card-text">Rocket : {rocket?.rocket_name}</p>
        <button className="btn btn-dark">Learn More</button>
      </div>
      {launch_success ? (
        <div className="badge fs-6 bg-success">Success</div>
      ) : (
        <div className="badge fs-6 bg-danger">Failure</div>
      )}
    </Link>
  );
};

export default Cards;
