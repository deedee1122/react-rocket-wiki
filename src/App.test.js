import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import App from "./App";
import Cards from "./components/Cards";
import Dummy from "./components/Dummy";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { MemoryRouter } from "react-router-dom";

test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/ /i);
  expect(linkElement).toBeInTheDocument();
});

test("Just a sample on how to test react componnets using snapshot testing", () => {
  const component = renderer.create(<Dummy></Dummy>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Cards
        links=""
        mission_name=""
        rocket=""
        launch_success=""
        flight_number=""
      />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
