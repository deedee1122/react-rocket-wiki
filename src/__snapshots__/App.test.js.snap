// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Just a sample on how to test react componnets using snapshot testing 1`] = `
<div>
  Dummy
</div>
`;

exports[`renders correctly 1`] = `
<div>
  Dummy
</div>
`;
